tests/storages_tests/test_storages.py

```diff
@@ -352,336 +452,406 @@ def test_get_trial_number_from_id(storage_mode):
+        non_existent_study_id = max(study_to_trials.keys()) + 1
+        with pytest.raises(KeyError):
+            storage.get_all_trials(non_existent_study_id)
 
-    storage = storage_init_func()
-    study_id = storage.create_new_study()
 
-    for trial in EXAMPLE_TRIALS:
-        _create_new_trial_with_example_trial(storage, study_id, EXAMPLE_DISTRIBUTIONS, trial)
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_get_all_trials_deepcopy_option(storage_mode: str) -> None:
 
-    with patch("copy.deepcopy", wraps=copy.deepcopy) as mock_object:
-        trials0 = storage.get_all_trials(study_id, deepcopy=False)
-        assert mock_object.call_count == 0
-        assert len(trials0) == len(EXAMPLE_TRIALS)
+    with StorageSupplier(storage_mode) as storage:
+        study_summaries, study_to_trials = _setup_studies(storage, n_study=2, n_trial=5, seed=49)
 
-        trials1 = storage.get_all_trials(study_id, deepcopy=True)
-        assert mock_object.call_count > 0
-        assert trials0 == trials1
+        for study_id in study_summaries:
+            with patch("copy.deepcopy", wraps=copy.deepcopy) as mock_object:
+                trials0 = storage.get_all_trials(study_id, deepcopy=True)
+                assert mock_object.call_count > 0
+                assert len(trials0) == len(study_to_trials[study_id])
 
+            # Check modifying output does not break the internal state of the storage.
+            trials0_original = copy.deepcopy(trials0)
+            trials0[0].params["x"] = 0.1
 
-@parametrize_storage
-def test_get_n_trials(storage_init_func):
-    # type: (Callable[[], BaseStorage]) -> None
+            with patch("copy.deepcopy", wraps=copy.deepcopy) as mock_object:
+                trials1 = storage.get_all_trials(study_id, deepcopy=False)
+                assert mock_object.call_count == 0
+                assert trials0_original == trials1
 
-    storage = storage_init_func()
-    study_id = storage.create_new_study()
 
-    _create_new_trial_with_example_trial(
-        storage, study_id, EXAMPLE_DISTRIBUTIONS, EXAMPLE_TRIALS[0]
-    )
-    _create_new_trial_with_example_trial(
-        storage, study_id, EXAMPLE_DISTRIBUTIONS, EXAMPLE_TRIALS[1]
-    )
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_get_n_trials(storage_mode: str) -> None:
+
+    with StorageSupplier(storage_mode) as storage:
+        study_id_to_summaries, _ = _setup_studies(storage, n_study=2, n_trial=7, seed=50)
+        for study_id in study_id_to_summaries.keys():
+            assert storage.get_n_trials(study_id) == 7
 
-    assert 2 == storage.get_n_trials(study_id)
-    assert 1 == storage.get_n_trials(study_id, TrialState.COMPLETE)
+        non_existent_study_id = max(study_id_to_summaries.keys()) + 1
+        with pytest.raises(KeyError):
+            assert storage.get_n_trials(non_existent_study_id)
 
 
 def _create_new_trial_with_example_trial(storage, study_id, distributions, example_trial):
```

Comment:
I think this function is not used and can be removed.

Reply:
Thank you. Removed.